### PR TITLE
Provider-agnostic authentication

### PR DESCRIPTION
**Pull request by houzhenyu1117**
Because Blitz now uses UAA as well as third-party identity providers (Google, Facebook), the profile objects returned by api/1/account/about can vary, and this results in failed authentication for users without an email address. This PR fixes logging in and starting blitz/rushes for all identity providers by using the entire profile object rather than individual fields like email.

Also included:
Fixes rush first empty results break code
Fixes links to results

### CODE REVIEW

**Comment by johntron** [comment]
@leonoraEAE same as https://github.com/blitz-io/blitz-chrome/pull/3 - this is ready for review. All the bugs have been fixed, so we just need to merge this in and then publish the plugin.

File: data/blitz/blitz.chrome.js [hunk 1]
```diff
@@ -181,14 +181,14 @@ $(function () {
...
-                    self.port.emit('abort', [user_id, api_key, job._id]);
+                    self.port.emit('abort', [profile, api_key, job._id]);
                     $(this).remove();
                 });
                 test.results = [];
                 test.$progress.attr('value', '0');
             }
 
-            if (job.result && job.result.timeline) {
+            if (job.result && job.result.timeline && job.result.timeline.length) {
```
Comment (leonoraEAE) [inline data/blitz/blitz.chrome.js]: Same as before, probably it would be best to test length > 0 given the subtraction below.

File: lib/main.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 const { Cc, Ci } = require('chrome');
-var widgets = require('sdk/widget');
+//var widgets = require('sdk/widget');
```
Comment (leonoraEAE) [inline lib/main.js]: remove comment ?

**Comment by leonoraEAE** [comment]
Review done. Just some minor comments.

**Pull request merged**